### PR TITLE
refine ZK-4791 (test detach parent then child, which means detach the…

### PR DESCRIPTION
**Pull request by DevChu**
… detached component)

### CODE REVIEW

**Review by scribetw — CHANGES_REQUESTED** [review]
wait for 4778

File: zkbind/src/org/zkoss/bind/tracker/impl/BindUiLifeCycle.java [hunk 1]
```diff
@@ -159,6 +159,9 @@ private boolean reInitBinder0(Component comp) {
 			return false;
 		}
 
+		// ZK-4791, reinit bindings
+		BindComposer.storeVmIdBinderRelation(comp);
```
Comment (DevChu) [inline zkbind/src/org/zkoss/bind/tracker/impl/BindUiLifeCycle.java:163]: move to util

File: zkbind/src/org/zkoss/bind/impl/BinderImpl.java [hunk 2]
```diff
@@ -2076,6 +2078,16 @@ protected ParamCall createParamCall(BindContext ctx) {
 	protected void doExecute(Component comp, String command, Map<String, Object> commandArgs, BindContext ctx,
 			Set<Property> notifys) {
 		try {
+			Matcher matcher = Pattern.compile("\\$([^.]*)\\..*$").matcher(command);
```
Comment (DevChu) [inline zkbind/src/org/zkoss/bind/impl/BinderImpl.java:2081]: move to private Pattern
"\\$([^.]+)\\..*"

File: zkbind/src/org/zkoss/bind/BindComposer.java [hunk 3]
```diff
@@ -657,4 +651,17 @@ public boolean service(AuRequest request, boolean everError) {
 		return false;
 	}
 
+	/**
+	 * Internal used only
+	 */
+	public static void storeVmIdBinderRelation(Component comp) {
+		Desktop desktop = comp.getDesktop();
```
Comment (DevChu) [inline zkbind/src/org/zkoss/bind/BindComposer.java:658]: null safe

File: zkbind/src/org/zkoss/bind/impl/BinderImpl.java [hunk 4]
```diff
@@ -259,6 +261,9 @@ public boolean isDefaultMethod(Method method) {
 	//ZK-3133 to cache MatchMedia annotation values
 	private transient Map<String, Method> _matchMediaValues;
 
+	//ZK-4791
+	private Pattern _callOtherVMCommandPattern = Pattern.compile("\\$([^.]*)\\..*$");
```
Comment (scribetw) [inline zkbind/src/org/zkoss/bind/impl/BinderImpl.java:265]: static final

**Review by scribetw — APPROVED** [review]

**Review by plijyfes — APPROVED** [review]

**Review by scribetw — APPROVED** [review]

**Pull request merged**